app/core/social-handlers/GoogleClassroomHandler.js

```diff
@@ -0,0 +1,132 @@
+import api from 'core/api'
+import CocoClass from 'core/CocoClass'
+
+const GoogleClassroomAPIHandler = class GoogleClassroomAPIHandler extends CocoClass {
+  
+  constructor () {
+    application.gplusHandler.loadAPI()
+    super()
+  }
+
+  loadClassroomsFromAPI () {
+    return new Promise((resolve, reject) => {
+      gapi.client.load ('classroom', 'v1', () => {
+        gapi.client.classroom.courses.list({access_token: application.gplusHandler.token(), teacherId: me.get('gplusID'), courseStates: "ACTIVE"})
+        .then((r) => {
+          resolve(r.result.courses || [])
+          })
+        .catch ((err) => {
+          console.error("Error in fetching from Google Classroom:", err)
+          reject(err)
+          })
+        })
+      })
+  }
+
+  loadStudentsFromAPI (googleClassroomId) {
+    return new Promise((resolve, reject) => {
+      gapi.client.load ('classroom', 'v1', () => {
+        gapi.client.classroom.courses.students.list({access_token: application.gplusHandler.token(), courseId: googleClassroomId})
+        .then((r) => {
+          resolve(r.result.students || [])
+          })
+        .catch ((err) => {
+          console.error("Error in fetching from Google Classroom:", err)
+          reject(err)
+          })
+        })
+      })
+  }
+
+}
+
+module.exports = {
+  gcApiHandler: new GoogleClassroomAPIHandler(),
+
+  scopes: 'https://www.googleapis.com/auth/classroom.courses https://www.googleapis.com/auth/classroom.courses.readonly https://www.googleapis.com/auth/classroom.profile.emails https://www.googleapis.com/auth/classroom.profile.photos https://www.googleapis.com/auth/classroom.rosters https://www.googleapis.com/auth/classroom.rosters.readonly',
+
+  markAsImported: async function(gcId) {
+    try {
+      let gClass = me.get('googleClassrooms').find((c)=>c.id==gcId)
+      if (gClass) {
+        gClass.importedToCoco = true
+        await new Promise(me.save().then)
```

Comment:
Is this passing the `then` chain method into the resolve method of the new Promise?
This is really cool.

Reply:
Yeah :)